test/built-ins/Temporal/ZonedDateTime/prototype/with/order-of-operations.js

```diff
@@ -0,0 +1,138 @@
+
+const expected = [
+  // RejectObjectWithCalendarOrTimeZone
+  "get fields.calendar",
+  "get fields.timeZone",
+  // CalendarFields / PrepareTemporalFields
+  "get this.calendar.fields",
+  "call this.calendar.fields",
+  "get fields.day",
+  "get fields.day.valueOf",
+  "call fields.day.valueOf",
+  "get fields.hour",
+  "get fields.hour.valueOf",
+  "call fields.hour.valueOf",
+  "get fields.microsecond",
+  "get fields.microsecond.valueOf",
+  "call fields.microsecond.valueOf",
+  "get fields.millisecond",
+  "get fields.millisecond.valueOf",
+  "call fields.millisecond.valueOf",
+  "get fields.minute",
+  "get fields.minute.valueOf",
+  "call fields.minute.valueOf",
+  "get fields.month",
+  "get fields.month.valueOf",
+  "call fields.month.valueOf",
+  "get fields.monthCode",
+  "get fields.monthCode.toString",
+  "call fields.monthCode.toString",
+  "get fields.nanosecond",
+  "get fields.nanosecond.valueOf",
+  "call fields.nanosecond.valueOf",
+  "get fields.second",
+  "get fields.second.valueOf",
+  "call fields.second.valueOf",
+  "get fields.year",
+  "get fields.year.valueOf",
+  "call fields.year.valueOf",
+  "get fields.offset",
+  "get fields.offset.toString",
+  "call fields.offset.toString",
+  // options
+  "get options.disambiguation",
+  "get options.disambiguation.toString",
+  "call options.disambiguation.toString",
+  "get options.offset",
+  "get options.offset.toString",
+  "call options.offset.toString",
+  // PrepareTemporalFields
+  "get this.timeZone.getOffsetNanosecondsFor",
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.calendar.day",
+  "call this.calendar.day",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.hour
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.microsecond
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.millisecond
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.minute
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.calendar.month",
+  "call this.calendar.month",
+  "get this.timeZone.getOffsetNanosecondsFor",
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.calendar.monthCode",
+  "call this.calendar.monthCode",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.nanosecond
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.second
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.timeZone.getOffsetNanosecondsFor",
+  "call this.timeZone.getOffsetNanosecondsFor",
+  "get this.calendar.year",
+  "call this.calendar.year",
+  "get this.timeZone.getOffsetNanosecondsFor",  // ZonedDateTime.p.offset
+  "call this.timeZone.getOffsetNanosecondsFor",
```

Comment:
I didn't understand how this could result from PrepareTemporalFields - nor could I replicate it manually tracing through user callable sites of the proposal text. What's going on here, and why are there so many calls expected to getOffsetNansecondsFor - am I missing something in the proposal text or is this an oops? Running it against the polyfill and engines shows it passes, so I'm guessing I'm just missing it and pardon the perhaps silly question.

Reply:
It's tricky. This occurs in step 14 of [Temporal.ZonedDateTime.prototype.with](https://tc39.es/proposal-temporal/#sec-temporal.zoneddatetime.prototype.with), "PrepareTemporalFields(_zonedDateTime_, _fieldNames_, ...)". That performs a Get on the ZonedDateTime object for each of the properties named in _fieldNames_. The property getters (e.g., [year](https://tc39.es/proposal-temporal/#sec-get-temporal.zoneddatetime.prototype.year)) each call [BuiltinTimeZoneGetPlainDateTimeFor](https://tc39.es/proposal-temporal/#sec-temporal-builtintimezonegetplaindatetimefor), which produces the getOffsetNanosecondsFor call.

Converting the exact time to a PlainDateTime so many times is a bit over the top, but it is what is specified. I think it can be addressed in https://github.com/tc39/proposal-temporal/issues/2289.

Reply:
Thank you very much! I forgot that "Get" wouldn't just call user code, but would also call the built-in getters. That was the missing link in my trace.